lib/ansible/modules/windows/win_iis_webhandler.ps1

```diff
@@ -0,0 +1,254 @@
+                        $result.changed = $true
+                    }
+                    catch {
+                        $ErrorMessage = "Failed to set desired state for existing webhandler's '$name' for property '$keyName' in IIS on '$iispath_for_messages' path: $($_.Exception.Message)"
+                        Fail-Json -obj $result -message $ErrorMessage
+                    } 
+                }
+                else {
+                    $in_desired_state = $false
+                    try {
+                        if ($iispath -eq "IIS:\" -and !$location) {
+                            Clear-WebConfiguration -Filter "$filter/@$keyName" -PSPath $iispath -WhatIf:$check_mode
+                        }
+                        else {
+                            Clear-WebConfiguration -Filter "$filter/@$keyName" -PSPath "IIS:\" -Location $Location -WhatIf:$check_mode
+                        }
+
+                        Add-Warning -obj $result -message "Existing webhandler's '$name' property '$keyName' was cleared in IIS on '$iispath_for_messages' path to be in desired state."
+                        $result.changed = $true
+                    }
+                    catch {
+                        $ErrorMessage = "Failed to clear existing webhandler's '$name' property '$keyName' in IIS on '$iispath_for_messages' path: $($_.Exception.Message)"
+                        Fail-Json -obj $result -message $ErrorMessage
+                    }
+                }
+            }
+        }
+
+        if ($in_desired_state -eq $true) {
+            Add-Warning -obj $result -message "Existing webhandler '$name' in IIS on '$iispath_for_messages' path already in desired state."
+        }
+    }
+}
+else {
+    if ($currentwebhandler) {
+        try {
+            if ($iispath -eq "IIS:\" -and !$location) {
+                Remove-WebHandler -Name $name -PSPath $iispath -Confirm:$false -WhatIf:$check_mode
+            }
+            else {
+                Remove-WebHandler -Name $name -PSPath "IIS:\" -Location $Location -Confirm:$false -WhatIf:$check_mode
+            }
+
+            Add-Warning -obj $result -message "Webhandler '$name' was removed from IIS on '$iispath_for_messages' path."
+            $result.changed = $true
+        }
+        catch {
+            $ErrorMessage = "Failed to remove webhandler '$name' from IIS on '$iispath_for_messages' path: $($_.Exception.Message)"
+            Fail-Json -obj $result -message $ErrorMessage
+        }
+    }
+    else {
+        Add-Warning -obj $result -message "Webhandler '$name' is already removed from IIS on '$iispath_for_messages' path."
+    }
+}
+
+# Result
+try {
+    $webhandler = Get-WebHandler -Name $name -PSPath $iispath
+
+}
+catch {
+    $ErrorMessage = "Failed to get info about webhandler '$name' in IIS on '$iispath_for_messages' path (during creating the result object): $($_.Exception.Message)"
+    Fail-Json -obj $result -message $ErrorMessage
```

Comment:
What happens when state is absent or we are in check mode and "created" a new handler? Doesn't seem like we should be failing here?

Reply:
Should be fixed now.